src/commands/deploy/nextjs/deploy.ts

```diff
@@ -46,13 +46,12 @@ import {
 import { readOrAskConfig } from "../utils.js";
 import { SSRFrameworkComponentType } from "../../../models/projectOptions.js";
 import { addSSRComponentToConfig } from "../../analyze/utils.js";
-
 export async function nextJsDeploy(options: GenezioDeployOptions) {
     const genezioConfig = await readOrAskConfig(options.config);
     const packageManagerType = genezioConfig.nextjs?.packageManager || NODE_DEFAULT_PACKAGE_MANAGER;
 
-    const cwd = process.cwd();
-    const componentPath = genezioConfig.nextjs?.path
+    let cwd = process.cwd();
+    let componentPath = genezioConfig.nextjs?.path
```

Comment:
These should not be defined with `let` because it will become confusing - we will start using this `cwd` in-between declarations and it will not be clear when it's `process.cwd` and when is not.

IMO:
- `cwd` should be where `genezio.yaml` is and where `genezio deploy` was ran, 
- `componentPath` is the relative path between `cwd` and the nextjs folder - the componentPath is saved in yaml at `next.path` and it's where we are running scripts and commands such as `npm install`.

For the `temp` folder processing and building I think you should use new vars named  something like`tempBuildCwd` and `tempBuildComponentPath`.